ocaml/testsuite/tests/typing-zero-alloc/signatures.ml

```diff
@@ -0,0 +1,821 @@
+(* TEST
+   * expect
+*)
+
+(* This tests the typing behavior of `[@zero_alloc]` annotation in signatures.
+
+   These tests are just about what is allowed and not allowed by the
+   typechecker.  The implementation of the actual `[@zero_alloc]` backend checks
+   (including how the annotations in signatures affect those checks) are tested
+   in the `tests/backend/checkmach` directory at the root of the project.
+*)
+
+(*******************************************)
+(* Test 1: Allowed and disallowed payloads *)
+module type S1_1 = sig
```

Comment:
I agree the numbers are a bit annoying.  On the other hand, I strongly dislike when the test programs use the same name repeatedly - I find it makes them hard to navigate, and have occasionally spent time staring at the wrong test.  And I'm bothered less by the numbers getting a bit wonky, since in my mind they exist just to be unique.

In any case, here it's reasonably easy to come up with mostly unique names, which seems obviously better, so I've done that.  (But I'd like to stop short of promising never to do this again if it proves harder to generate good names in a future case)